### ISSUE TITLE
Clojure 1.8 support broken with CI update

### ISSUE THREAD

**Issue by bm3719** [clojure-emacs/cider-nrepl#864]
Following the change in ea56343 by @vemv, it's no longer possible to load cider-nrepl on any project using Clojure <1.9.

I understand wanting to sync the Circle CI tests to supported versions, and it'll currently display a warning accordingly.  But, also completely preventing cider-nrepl from being loaded will mean we can no longer use it at all for any projects using 1.8.

Seems like having the logic in `cider-nrepl.plugin` still allow it to load (with warning) would be more permissive and not lock out as many old projects without incurring any downsides.

**Comment by vemv** [comment]
Hi @bm3719 ,

Per https://github.com/clojure-emacs/cider-nrepl/pull/840 :

> 1.8 compat, while it was intended to be, was broken for a possibly very long time.

We also intended to drop it intentionally anyway. So, it was unawarely broken, and we did want to drop its support anyway.

It's not wortwhile to support that version with Clojure 1.12 around the corner.

Is there a strong reason why you'd use Clojure 1.8?

We certainly do not require you to bump the Clojure version for production usage:  it's east enough to bump Clojure  locally, in a `:dev`/`:cider-nrepl` Lein profile or t.deps alias.

Cheers - V
